File: x_verba/_test_sample/sample.go
package agent

import (
	"context"
	"fmt"

	"github.com/sashabaranov/go-openai"
)

func RunAgent(ctx context.Context, input string, mode string) (string, error) {
	isValid := validateInput(input)
	if !isValid {
		return "", fmt.Errorf("invalid input")
	} else if mode == "fast" {
		fmt.Println("fast mode")
	}

	switch mode {
	case "fast":
		doFastThing()
	case "slow":
		doSlowThing()
	default:
		doDefaultThing()
	}

	for i := 0; i < 3; i++ {
		fmt.Println(i)
	}

	result, err := fetchData(ctx, input)
	if err != nil {
		return "", err
	}

	data, err := db.Delete(input)
	if err != nil {
		log.Error("delete failed", err)
		notifyAdmin(err)
	}

	if requireApproval(input) {
		agent.Invoke(input)
	}

	client := openai.NewClient("sk-test")
	resp, err := client.CreateChatCompletion(ctx, openai.ChatCompletionRequest{
		Model:       "gpt-4",
		Temperature: 0.7,
		MaxTokens:   500,
	})
	if err != nil {
		return "", err
	}

	return resp.Choices[0].Message.Content, nil
}

File: x_verba/_test_sample/sample.rs
use async_openai::Client;

fn run_agent(input: &str, mode: &str) -> String {
    let is_valid = validate_input(input);
    if !is_valid {
        panic!("invalid input");
    } else if mode == "fast" {
        println!("fast mode");
    }

    match mode {
        "fast" => do_fast_thing(),
        "slow" => do_slow_thing(),
        _ => do_default_thing(),
    }

    for item in 0..3 {
        println!("{}", item);
    }

    let data = fetch_data(input).unwrap();

    let result = db_delete(input).expect("delete failed");

    if require_approval(input) {
        agent.invoke(input);
    }

    let client = Client::new();
    let response = client.chat().create(request).unwrap();

    response
}
